scripts/zones/Maze_of_Shakhrami/mobs/Leech_King.lua

```diff
@@ -0,0 +1,23 @@
+-----------------------------------
+-- Area: Maze of Shakhrami
+--  MOB: Leech King (NM)
+-----------------------------------
+
+require("scripts/zones/Maze_of_Shakhrami/MobIDs");
+
+-----------------------------------
+-- onMobDeath
+-----------------------------------
+
+function onMobDeath(mob,killer,ally)
+    local whichNM = math.random(0,19);
+
+    if (whichNM < 10) then
+        SetRespawnTime(Argus, 64800, 108000); -- 18-30 hours
+        DeterMob(Leech_King, true);
+    else
+        SetRespawnTime(Leech_King, 64800, 108000); -- 18-30 hours
+        DeterMob(Argus, true);
```

Comment:
@Hozu my idea was to always deter one mob, so there is no way they will be up at the same time

I can change the logic if it's the better/prefered way, just thought those calls 2 calls every 12+ hours wouldn't matter much :)